internal/app/distro_versions.go

```diff
@@ -0,0 +1,206 @@
+	if err != nil {
+		return releases, fmt.Errorf("error getting latest distro version: %w", err)
+	}
+
+	// Calculate the latest supported version based on the latest release.
+	latestSupportedVersion := GetLatestSupportedVersion(latestRelease.Version)
+
+	// Loop over all tags except the final element and only keep supported ones.
+	for _, tag := range tags {
+		v, err := VersionFromRef(tag.Ref)
+		if err != nil || v.LessThan(&latestSupportedVersion) || v.Prerelease() != "" {
+			continue
+		}
+
+		release, err := newDistroRelease(ghClient, tag)
+		if err != nil {
+			// Skip tags that cannot be parsed or processed.
+			continue
+		}
+
+		releases = append(releases, release)
+	}
+
+	slices.Reverse(releases)
+
+	return releases, nil
+}
+
+const previousSupportedVersions = 2
+
+// GetLatestSupportedVersion returns the supported version based on the second segment of the version.
+func GetLatestSupportedVersion(v version.Version) version.Version {
+	// Generate a version string using the second segment from the provided version.
+	versionStr := fmt.Sprintf("1.%d.0", v.Segments()[1]-previousSupportedVersions)
+
+	supportedVersion, err := version.NewSemver(versionStr)
+	if err != nil {
+		return version.Version{}
+	}
+
+	return *supportedVersion
+}
+
+var (
+	ErrLatestDistroVersionNotFound = errors.New("latest distro not found")
+	ErrInvalidVersion              = errors.New("invalid version")
+)
+
+// GetLatestDistroVersion iterates backward over tags to return the latest valid distro release.
+func getLatestDistroVersion(ghClient git.RepoClient, tags []git.Tag) (DistroRelease, error) {
+	// Iterate from last to first using slices.Backward.
+	for _, tag := range slices.Backward(tags) {
+		version, err := VersionFromRef(tag.Ref)
+		if err != nil {
+			continue
+		}
+		// Skip prerelease versions.
+		if version.Prerelease() != "" {
+			continue
+		}
+
+		return newDistroRelease(ghClient, tag)
+	}
+
+	return DistroRelease{}, ErrLatestDistroVersionNotFound
+}
+
+// NewDistroRelease creates a DistroRelease from a Tag, fetching its commit details.
+func newDistroRelease(ghClient git.RepoClient, tag git.Tag) (DistroRelease, error) {
+	var release DistroRelease
+
+	// Parse version from tag reference.
+	version, err := VersionFromRef(tag.Ref)
+	if err != nil {
+		logrus.Debug(err)
+
+		return release, fmt.Errorf("invalid version: %w", err)
+	}
+
+	// Fetch the commit information using the URL.
+	commit, err := ghClient.GetObjectInfo(tag.Object.URL)
+	if err != nil {
+		logrus.Error(err)
+
+		return release, fmt.Errorf("error getting commit: %w", err)
+	}
+
+	// Parse the commit date.
+	var commitDate time.Time
+	if commit.Author != nil {
+		commitDate, err = time.Parse(time.RFC3339, commit.Author.Date)
+		if err != nil {
+			commitDate = time.Time{}
+		}
+	} else if commit.Tagger != nil {
+		commitDate, err = time.Parse(time.RFC3339, commit.Tagger.Date)
+		if err != nil {
+			commitDate = time.Time{}
+		}
+	}
```

Comment:
You are using the zero value (`0001-01-01 00:00:00 +0000 UTC`) here as a fallback but I don't see logic in the presentation to show something more relevant to the user instead of the zero time (may be I missed it).